hq_superset/hq_domain.py

```diff
@@ -51,8 +58,11 @@ def is_valid_user_domain(hq_domain):
 
 
 def user_domains(user):
-    # Todo Implement based on persisted user domain membership data
-    return ['a', 'b']
+    # This should be set by oauth_user_info after OAuth
+    return [
+        d["domain_name"]
+        for d in session["user_hq_domains"]["objects"]
```

Comment:
@proteusvacuum @akashkj @kaapstorm 
I have modified the select-domain methodology a bit. The user's superset enabled domains are stored in the session at login. Immediately after the login, the user gets redirected to a select domain page (unless there is only a single domain), the selected domain is set on the request cookie for the rest of the session. To select another domain, user would have to navigate to the select domain page or logout/login.

Reply:
Is there possibility of getting null error when there is no valid domain for the user?

Reply:
I like that approach